app/assets/javascripts/browsered/payment-request.js

```diff
@@ -0,0 +1,185 @@
+      } else {
+        merchantCapabilities = ['supports3DS']
+      }
+
+      return {
+        countryCode: 'GB',
+        currencyCode: details.total.amount.currency,
+        total: {
+          label: details.total.label,
+          amount: details.total.amount.value,
+        },
+        supportedNetworks: supportedNetworksAppleFormatted,
+        merchantCapabilities,
+        requiredShippingContactFields: ['email'],
+      }
+    } else {
+      return {
+        supportedInstruments,
+        details,
+        options
+      }
+    }
+  }
+
+  if (window.PaymentRequest && paymentMethodForm) {
+    paymentMethodForm.classList.remove('hidden')
+    standardMethodContainer.classList.add('hidden')
+
+    if (window.ApplePaySession && ApplePaySession.canMakePayments()) {
+      document.getElementById('payment-method-payment-request-apple').parentNode.style.display = 'block'
+    } else if (/Chrome/.test(navigator.userAgent) && /Google Inc/.test) {
+      document.getElementById('payment-method-payment-request').parentNode.style.display = 'block'
+    }
+
+    document.getElementById('payment-request-container').addEventListener('submit', function(e) {
+      e.preventDefault()
+      const checkedValue = document.querySelectorAll('#payment-request-container input:checked')[0].value
+      if (checkedValue === 'standard') {
+        standardMethodContainer.classList.remove('hidden')
+        paymentMethodForm.classList.add('hidden')
+      } else if (checkedValue === 'payment-request') {
+        makePaymentRequest()
+      } else if (checkedValue === 'payment-request-apple') {
+        makeApplePayRequest()
+      }
+    }, false)
+  }
+
+  function getApplePaySession(url) {
+    return fetch(`/apple-pay-merchant-validation`, {
+      method: 'POST',
+      credentials: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ url })
+    }).then(response => {
+      if (response.status >= 200 && response.status < 300) {
+        return response.json().then(data => {
+          return data
+        })
+      } else {
+        return session.abort();
+      }
+    })
+  }
+
+  function makeApplePayRequest() {
+    const session = new ApplePaySession(3, prepareRequestObject('apple'))
+
+    session.onvalidatemerchant = event => {
+      const validationURL = event.validationURL
+      getApplePaySession(event.validationURL).then(response => {
+        session.completeMerchantValidation(response)
+      }).catch(err => {
+        console.log('Couldn’t contact Apple Pay server', err)
+        return err
+      })
+    }
+
+    session.onpaymentauthorized = event => {
+      // Send payment for processing...
+      const payment = event.payment;
+      console.log('authorisation complete', payment)
+
+      // ...return a status and redirect to a confirmation page
+      session.completePayment(ApplePaySession.STATUS_SUCCESS);
+      // window.location.href = "/success.html";
```

Comment:
Should this be here?

Reply:
Double checked with Jon and this gets stripped out in production when it's minified.